api/v1alpha08/sonataflowplatform_services_types.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha08
+
+// ServicesPlatformSpec describes the desired service configuration for "prod" workflows.
+type ServicesPlatformSpec struct {
+	// Deploys the Data Index service for use by "prod" profile workflows.
+	// +optional
+	DataIndex *ServicesPodTemplateSpec `json:"dataIndex,omitempty"`
```

Comment:
pointer here is to reflect that if ServicesPodTemplateSpec = nil, no need deploy the data index?

Reply:
correct @wmedvede